generator/templates/base_struct_function.m

```diff
@@ -0,0 +1,66 @@
+{#- To avoid code duplication was crated this parent file, which contain common part used in:
+    "templates/functions/template.m" and "templates/structs/template.m". -#}
+{% include 'copyright.txt' %}
+{%- block imports %}
+#import "{{name}}.h"
+#import "NSMutableDictionary+Store.h"
+{%- for import in imports %}
+#import "{{import}}.h"
+{%- endfor %}
+{%- endblock %}
+
+NS_ASSUME_NONNULL_BEGIN
+
+@implementation {{name}}
```

Comment:
When implementing a deprecated class (such as `SetDisplayLayout`) you have to wrap the `@implementation` line with

```
#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Wdeprecated-implementations"

#pragma clang diagnostic pop
```

Reply:
all parameter should be wrapped together, or each parameter should be wrapped separately?

Reply:
done in https://github.com/smartdevicelink/sdl_ios/pull/1556/commits/97fc74359dc0ac2b615be9fb615416edb22becd9

Reply:
You wrapped it in `deprecated-declarations`, please see above for the correct diagnostic.

Reply:
```objc
#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Wdeprecated-declarations"
@implementation SDLSetDisplayLayout
#pragma clang diagnostic pop
```
or
```objc
#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Wdeprecated-declarations"
@implementation SDLSetDisplayLayout
...
@end
#pragma clang diagnostic pop

Reply:
done in https://github.com/smartdevicelink/sdl_ios/pull/1556/commits/8d538469711fe0690157629c5f8d0c361b29cdff